wagtail/contrib/forms/forms.py

```diff
@@ -47,6 +48,9 @@ def create_url_field(self, field, options):
     def create_number_field(self, field, options):
         return django.forms.DecimalField(**options)
 
+    def create_tel_field(self, field, options):
+        return django.forms.CharField(widget=TelephoneInput, **options)
```

Comment:
I really think a phone number field should provide a way to add validation so that only phone numbers for a particular country can be inserted.

Would it be possible to allow people to use ``PhoneNumberField`` from [``django-phonenumber-field``](https://github.com/stefanfoulis/django-phonenumber-field) here?

Reply:
Hi @kaedroho, that's exactly what I did originally, but if you look a bit earlier in this conversation, you'll see @gasman said that was too heavyweight for this purpose and that we don't need the validation, just the `type="tel"` attribute, and asked me to make a custom widget instead.

Reply:
> Thanks for the updates!
> For something that's only going to be used infrequently, the dependency on django-phonenumber-field is a bit heavyweight - we're only really interested in setting the `type="tel"` attribute, not the national phone number validation that python-phonenumbers gives us. I think it should be possible to do what we need by defining a custom widget type for type="tel" (this should only take a couple of lines, [like the ones built into Django](https://github.com/django/django/blob/2b4b6c8af0aae8785bc1347cf1be2e8e70fd5ff3/django/forms/widgets.py#L308-L310)) and passing that in create_tel_field (much like create_multiline_field does).

Reply:
Does `type="tel"` provide any validation if you don't specify a pattern? According to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/tel), it behaves the same as a text field on desktop, but on mobile phones it shows a keypad when you edit it.

Reply:
@kaedroho Yes, that's exactly what was requested in #6611.

Reply:
@gasman sorry I can't see any mention of using `pattern` there?

Reply:
@kaedroho as I understand it, it has nothing to do with `pattern` or validation. The problem was that  `<type="number">` was bringing up the wrong keyboard on mobile, so the addition of `<type="tel">` is just intended to make the correct keyboard display. That's all.

Reply:
If we provide a "telephone number" option, wouldn't user's expect that to include some sort of validation?

Reply:
I think that might be beyond the scope of what I can answer... Perhaps @gasman or @SharmaineLim (who created the issue #6611) can decide which approach is best. If you want validation, you can use https://github.com/wagtail/wagtail/pull/6723/commits/c8420ff62a09328bd1148d85af762ff5ec59aaee (which builds on changes made in https://github.com/wagtail/wagtail/pull/6723/commits/c249429d383ae44ee0311bebe9d8853a70ad80be), if you don't want validation, you can use https://github.com/wagtail/wagtail/pull/6723/commits/77bf795352f09e2f6271db97a969d97113f4317b.

Reply:
We discussed this in todays core team meeting. Here's an approach that we've come up with:

 - Add a setting to allow a regex pattern to be specified for phone number validation
 - Add a warning into the form builder docs and in the 2.13 upgrade considerations to warn developers that there will be no phone number validation if this setting isn't used

Let us know what you think! Sorry for all the back and forth on this

Reply:
Okay cool, will do!

Reply:
Hi @kaedroho I would like to finish this, but could I get some guidance on how to go about it please?